### ISSUE TITLE
PHP 7 private word

### ISSUE THREAD

**Issue by flug** [coduo/php-humanizer#20]
Hi, 
In PhP7 the word String by example is a protected word by native type

**Comment by norzechowicz** [comment]
Ah right, when humanizer was invented we werent aware of that, any suggestions how we could rename those classes?

**Comment by h4cc** [comment]
I would propose `String` becomes `HumanString` and accordingly if needed.

**Comment by norzechowicz** [comment]
``HumanString`` sounds good but its more a ``HumanStringTranslator`` rather than ``HumanString`` by itself. What do you think? 

This change is possible only if we dump humanizer version into 2.0.0 in order to follow semantic versioning rules.

**Comment by norzechowicz** [comment]
Or maybe ``StringHumanizer``, it would be like ``StringHumanizer::truncate($text, 8);``

**Comment by kelunik** [comment]
You can also rename it and use `class_alias` for backwards compatibility for PHP 5.

**Comment by idmontie** [comment]
:+1: this would be great. I vote `String` -> `HumanString`. If someone doesn't like how long it is, they can always alias it if they are using PHP 5 (`use Coduo\PHPHumanizer\HumanString as String`).

**Comment by norzechowicz** [comment]
problem solved, now I just need to tag version 2.0.0 but I would like to clean few things first.

**Comment by kelunik** [comment]
Why do you have to tag `2.0.0` for that? Just declare aliases on PHP 5 with `class_alias` to keep backwards compatibility.

**Comment by norzechowicz** [comment]
could you elaborate more about that?

**Comment by kelunik** [comment]
```php
if (PHP_VERSION_ID < 70000) {
    class_alias("Coduo\\PHPHumanizer\\HumanString", "Coduo\\PHPHumanizer\\String");
}
```

Something like that should keep old code on PHP 5 working while new code which should be compatible with PHP 7 can use `HumanString`. As `String` didn't work on PHP 7 ever, it's not a BC break there. Once you have enough other breaking features, you can remove the alias and only go with the new class name in `2.0.0`. Be sure to mark it as deprecated now either way.

**Comment by h4cc** [comment]
Nice solution :+1:

**Comment by norzechowicz** [comment]
How it would work for something like that in user code?

```php

use Coduo\PHPHumanizer\String;

class MyString extends String
{
//...
}
```

**Comment by kelunik** [comment]
That code continues to work, but doesn't work with PHP 7. PHP 7 users have to use `HumanString` everywhere.

**Comment by norzechowicz** [comment]
Yep, so its BC Breake and thats why we need to tag version 2.0.0. But version 2.0.0 is also a nice opportunity to close extension points like this one by adding final statement to classes that we wont like users to extend from in order to prevent such situations in future.

**Comment by kelunik** [comment]
No, you don't break any backwards compatibility, because it was never compatible with PHP 7 so far.

**Comment by norzechowicz** [comment]
yea bc breake is maybe not the best way to describe this situation, anyway adding support for version 7.0 with requirement to fix this edge case after update from php 5.6 to 7.0 seems like enough reason for me to tag this as a new version. 
Is there any reason why we should not release v2.0.0?
